apps/services/notifications/src/app/modules/notifications/consumer.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Inject, Injectable, OnModuleDestroy } from '@nestjs/common'
+import {
+  Message as SqsMessage,
+  ReceiveMessageCommand,
+  DeleteMessageBatchCommand,
+} from '@aws-sdk/client-sqs'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+import { CONNECTION_PROVIDER } from '../../../constants'
+import { SqsChannel } from '../../../types'
+import { MessageHandlerService } from './messageHandler.service'
+
+@Injectable()
+export class NotificationConsumerService implements OnModuleDestroy {
+  private currentBatchProcessing: Promise<void> = Promise.resolve()
+  private shutdown = false
+
+  constructor(
+    private readonly messageHandler: MessageHandlerService,
+    @Inject(CONNECTION_PROVIDER)
+    private readonly queue: SqsChannel,
+    @Inject(LOGGER_PROVIDER)
+    private readonly logger: Logger,
+  ) {}
+
+  async onModuleDestroy() {
+    this.shutdown = true
+
+    this.logger.info('Waiting for workers to finish')
+    await this.currentBatchProcessing
+
+    this.logger.info('Closing connection')
+    this.queue.client.destroy()
+  }
+
+  async run(): Promise<void> {
+    this.logger.info('Starting consumer')
+
+    while (!this.shutdown) {
+      const { Messages: messages = [] } = await this.queue.client.send(
+        new ReceiveMessageCommand({
+          QueueUrl: this.queue.queueUrl,
+          MaxNumberOfMessages: 10,
+          WaitTimeSeconds: 20,
```

Comment:
Maybe we should move these values to env variable for easy tuning in production?

Reply:
Maybe. Not sure.

These values are the maxmimum allowed.
If we make them configurable it would cause runtime errors to set them higher than SQS allows.

Reply:
Maybe that's not such a big deal since this is only the worker and it would not cause any real downtime/interruption